pkg/server/server.go

```diff
@@ -159,12 +167,65 @@ func createEnvironment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	//space delimited annotation of valid hostnames
+	var hostsList bytes.Buffer
+
+	//Loop through slice of HostNames
+	for index, value := range tempJSON.HostNames {
+		//Verify each Hostname matches regex
+		validIP := validIPAddressRegex.MatchString(value)
```

Comment:
This works, but you might want to add validation within the scope of the environmentPost to keep it encapsulated?  So that the caller would invoke like this

```
environmentPost.Validate()
```

This seems to be the standard pattern for go REST services I've found.